src/main/riddl/Venues/venueMessages.riddl

```diff
@@ -0,0 +1,79 @@
+command EstablishVenue is {
+  venueId: VenueId briefly "A unique identifier for this Venue. This must be provided by the calling client.",
+  info: VenueContext.Info briefly "Details about a venue required for it to exist. See VenueContext.Info for details.",
+  locations: LocationId*,
+  establishingMember: MemberId briefly "The Id of the Member who is issuing the create command.",
+  owner: MemberId? briefly "Used in the case that establishingMember is not the owner"
+} briefly "Creates a Venue with all top-level fields, potentially without Locations"
+described in file "establishVenue.md"
+event VenueEstablished is {
+  venueId: VenueId,
+  info: VenueContext.Info,
+  locations: LocationId*,
```

Comment:
I don't think Venues should have Locations

Reply:
Why? A location is something inside a Venue, right?  So, how did you want to express the relationship?

Reply:
Long story. But part of the complexity I have found while modeling this app is relationships. Venues has a collection of Locations. A Location belongs to a Venue. I think one "side" or the other of the relationship needs to own the data about the relationship. If both own the data, because of isolation, you run into a multi-master situation which is difficult to keep synchronized. It also makes logic in the handlers more complex. So pick one side of the relationship. I picked the Location side to own the Venue<->Location relationship. The other side of the relationship can be obtained by using a projection. This projection listens for LocationAdded events and updates the projection if the Location specifies a Venue.

Why did I pick the location side to "own" the relationship? In this case, it is the simple side of a 1:M relationship. I don't need to parse a list to figure out how to update the other side of the relationship. 

Another factor I am considering is an idea I am calling a "defining attribute". You need to maintain data in some way on both sides of a relationship - either as a field or within a projection. If the linking attribute is optional I would say that data is not defining. So that entity is a poor candidate for "owning" the relationship. In this specific case, a Venue must have at least one Location. A Location, on the other hand, does not need to specify a Venue. Using this logic Venue would be the better candidate to own the relationship because at least Location is required. 

I weighted the cardinality of the 1:M relationship and decided that modeling Location as the "owner" of the relationship made the overall model simpler. I am very open to more conversation on this as I think it is an important topic that will help future authors of RIDDL code considerably.

Reply:
If Location is 'defining attribute' of Venue-Location relationship, Venue's State relies on the projection of that relationship - as long as we are consistent with a requirement I came up with that a Venue cannot be outside Draft State without at least one Location.

Reply:
What I was trying to say is I think one side of the relationship needs to be authoritative for the data defining the relationship. I was trying to explain my logic for how I arrived at Location being that thing for the Venue<->Location relationship. I considered many things (including "defining attributes") but settled on the side of simplicity; the 1 side of the 1:M relationship to own the relationship data. I think the state transition rules would have to consult the projection to ensure we are in compliance.

I am still working these design ideas out in my mind. Keep challenging me. It helps sharpen those thoughts.